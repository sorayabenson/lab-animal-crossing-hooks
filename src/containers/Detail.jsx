import React from 'react';
import Spinner from '../components/Spinner';
import VillagerDetail from '../components/detail/VillagerDetail';
import { useDetail } from '../hooks/detail';


const Detail = () => {
    const {loading, villager} = useDetail();

    if (loading) return <Spinner />
    
    return (
        <main>
            <VillagerDetail {...villager}/>
        </main>
    )
}

export default Detail;   