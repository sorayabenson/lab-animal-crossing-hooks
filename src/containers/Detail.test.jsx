import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import Detail from './Detail';
import { MemoryRouter, Route } from 'react-router-dom';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import mockDetailData from '../../fixtures/detail.json';

const server = setupServer(
    rest.get(
        'https://acnhapi.com/v1/villagers/46',
        (req, res, ctx) => {
            return res(
                ctx.json(mockDetailData)
            )
        })
)

describe('Detail container', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());
    
    it('renders Detail', async () => {
        render(
            <MemoryRouter initialEntries={['/46']}>
                <Route path='/:id'>
                    <Detail />
                </Route>
            </MemoryRouter>
        );

        screen.getByText('circling for landing');
        
        await waitFor(() => {
            const image = screen.getAllByAltText('Kiki');
            expect(image).toMatchSnapshot();            
            const name = screen.getByText('Kiki');
            expect(name).toMatchSnapshot();
            const catchphrase = screen.getByText('kitty cat');
            expect(catchphrase).toMatchSnapshot(); 
        })
    })
})