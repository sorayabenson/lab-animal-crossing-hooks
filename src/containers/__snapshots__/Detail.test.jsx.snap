// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail container renders Detail 1`] = `
Array [
  <img
    alt="Kiki"
    src="https://acnhapi.com/v1/images/villagers/46"
  />,
]
`;

exports[`Detail container renders Detail 2`] = `
<h2>
  Kiki
</h2>
`;

exports[`Detail container renders Detail 3`] = `
<h3>
  "
  kitty cat
  "
</h3>
`;
